__tests__/version.test.ts

```diff
@@ -0,0 +1,239 @@
+/**
+ * Unit tests for src/version.ts
+ */
+import { jest } from "@jest/globals"
+import nock from "nock"
+
+import * as core from "../__fixtures__/core.js"
+import { testVersions, versionsJsonFile } from "../__fixtures__/constants.js"
+
+// Mocks should be declared before the module being tested is imported.
+jest.unstable_mockModule("@actions/core", () => core)
+
+// The module being tested should be imported dynamically. This ensures that the
+// mocks are used in place of any actual dependencies.
+const { resolveVersions, resolveVersion, versionSort } = await import(
+  "../src/version.js"
+)
+
+describe("versionSort tests", () => {
+  it("Returns the proper order", () => {
+    const arr = ["5.5.1", "4.21.0", "4.22.0", "6.1.0", "5.1.0", "4.5.0"]
+    const expected = ["4.5.0", "4.21.0", "4.22.0", "5.1.0", "5.5.1", "6.1.0"]
+
+    expect(arr).not.toEqual(expected)
+    expect(arr.sort()).not.toEqual(expected)
+    expect(versionSort(arr)).toEqual(expected)
+  })
+})
+
+describe("resolveVersions tests", () => {
+  beforeEach(() => {
+    // Instead of downloading versions.json, use `__fixtures__/versions.json`.
+    // Mocking `node-fetch` with `jest` directly or `jest-fetch-mock` doesn't work well with ESM:
+    // https://github.com/node-fetch/node-fetch/issues/1263
+    nock("https://julialang-s3.julialang.org")
+      .persist()
+      .get("/bin/versions.json")
+      .replyWithFile(200, versionsJsonFile)
+
+    // Replicate the available nightlies when versions.json was downloaded.
+    nock("https://julialangnightlies-s3.julialang.org")
+      .persist()
+      .head(/1\.(10|11|12)/)
+      .reply(200)
+      .head(/.*/)
+      .reply(404)
+  })
+
+  afterEach(() => {
+    jest.resetAllMocks()
+    nock.cleanAll()
+  })
+
+  it("Handles nightly", async () => {
+    await expect(resolveVersions(["nightly"], ".")).resolves.toEqual([
+      "nightly",
+    ])
+    await expect(resolveVersions(["1.10-nightly"], ".")).resolves.toEqual([
+      "1.10-nightly",
+    ])
+    await expect(resolveVersions(["1.11-nightly"], ".")).resolves.toEqual([
+      "1.11-nightly",
+    ])
+    await expect(resolveVersions(["1.12-nightly"], ".")).resolves.toEqual([
+      "1.12-nightly",
+    ])
+  })
+
+  it("Respects ifMissing error", async () => {
+    await expect(
+      resolveVersions(["1.9-nightly"], ".", { ifMissing: "error" })
+    ).rejects.toThrow("No Julia version exists")
+  })
+
+  it("Respects ifMissing warning", async () => {
+    await resolveVersions(["1.9-nightly"], ".", { ifMissing: "warn" })
+    expect(core.warning).toHaveBeenNthCalledWith(
+      1,
+      expect.stringMatching(/^No Julia version exists/)
+    )
+  })
+
+  it("Default ifMissing behavior", async () => {
+    await expect(resolveVersions(["1.9-nightly"], ".")).rejects.toThrow(
+      "No Julia version exists"
+    )
+  })
+})
+
+describe("resolveVersion tests", () => {
+  // Will need to update these values if `__fixtures__/versions.json` is updated.
```

Comment:
A note in the `CONTRIBUTORS` guidelines would be useful so this doesn't get lost. I personally also preface these kinds of things w/ `TODO` so when I search a repo I can easily find similar notes.

Reply:
I'll add these details into the CONTRIBUTORS documentation but I think a TODO is out of place as it isn't something that can be finished. It's more of a warning.